### DISCUSSION TITLE
Is there a way to detect header and footer coordinates?

### DISCUSSION THREAD

**Discussion by ghost** [pymupdf/PyMuPDF#1804] [Q&A] [answered]
In the script I'm working with I need to redact these parts before getting the text from the document because it breaks up the text and I have no way to regex it out because I need this script to work with different documents. Is there a way to find the footer / header coordinates to put them in a Rect and redact?

**Comment by JorjMcKie** [comment, marked as answer]
PDF knows nothing about such things as "header" or "footer".
You have to rely on "outside" knowledge.
But once you have the resp. bboxes, you can redact away those text contents of course.
So either you know beforehand that doc type x has headers/footers at positions y and z, or you must extract the full text first and do whatever heuristics to find and eliminate those parts from the extraction output.
In the latter case you obviously no more need to redact anything.
